packages/server/src/service/pdfForm.ts

```diff
@@ -97,3 +104,110 @@ export const fillNewHampshire = (
   }
 )
+
+// Return [street address, city, state, zip code]
+function splitAddress(address: string) {
+  // We're going to have a problem here if someone puts a commma
+  // between street address and apartment number.
+  const streetAddress = address.split(',')[0]
+  const city = address.split(',')[1].trim()
+  const state = address.split(',')[2].split(' ')[1]
+  const zipCode = address.split(',')[2].split(' ')[2]
+  return [streetAddress, city, state, zipCode]
+}
+
+export const fillNorthCarolina = (
+  stateInfo: NorthCarolinaInfo
+) => fillFormWrapper(
+  __dirname + '/forms/North_Carolina.pdf',
+  async ({check, text, textWithSize, placeImage}) => {
+    const nameSplit = stateInfo.name.split(' ')
+    let firstName = ''
+    let lastName = ''
+    let middleName = ''
+    let suffix = ''
+    firstName = nameSplit[0]
+    if(nameSplit.length == 2) {
+      lastName = nameSplit[1]     
+    } else if(nameSplit.length == 3) {
+      if(nameSplit[2].length < 4 
+         && (nameSplit[2].toLowerCase().includes('jr')
+         || nameSplit[2].toLowerCase().includes('sr'))) {
+           lastName = nameSplit[1]
+           suffix = nameSplit[2]
+         } else {
+           lastName = nameSplit[2]
+           middleName = nameSplit[1]
+         }
+    } else if(nameSplit.length == 4) {
+      middleName = nameSplit[1]
+      lastName = nameSplit[2]
+      suffix = nameSplit[3]
+    } else {
+      throw new Error('Cannot handle this name.')
+    }
+    text(lastName, 2, 56, 90)
+    text(firstName, 2, 226, 90)
+    text(middleName, 2, 355, 90)
+    text(suffix, 2, 440, 90)
+
+    const birthdateSplit = stateInfo.birthdate.split('-')
+    // Month
+    text(birthdateSplit[1], 2, 481, 90)
+    // Day
+    text(birthdateSplit[2], 2, 508, 90)
+    // Year
+    text(birthdateSplit[0], 2, 542, 90)
+
+    if(stateInfo.idType.includes('SSN')) {
+      text(stateInfo.idData[0], 2, 497, 134)
+      text(stateInfo.idData[1], 2, 520, 134)
+      text(stateInfo.idData[2], 2, 543, 134)
+      text(stateInfo.idData[3], 2, 566, 134)
+    } else {
+      text(stateInfo.idData, 2, 370, 134)
+    }
+
+    const uspsSplit = splitAddress(stateInfo.uspsAddress)
+    text(uspsSplit[0], 2, 60, 175)
+    text(uspsSplit[1], 2, 55, 207)
+    textWithSize(uspsSplit[2], 2, 134, 207, 9)
+    textWithSize(uspsSplit[3], 2, 158, 207, 9)
+
+    if(stateInfo.dateMoved) {
+      check(2, 396, 170)
+      const dateMovedSplit = stateInfo.dateMoved.split('-')
+      // Month
+      text(dateMovedSplit[1], 2, 311, 198)
+      // Day
+      text(dateMovedSplit[2], 2, 338, 198)
+      // Year
+      text(dateMovedSplit[0], 2, 362, 198)
+    } else {
+      check(2, 375, 170)
+    }
+
+    // Get county data from contact.
+    text(stateInfo.contact.county ? stateInfo.contact.county.split(' ')[0] : '', 2, 200, 207)
+
+    check(2, 498, 272) // General election
+
+    // Sic: we want 'Same as above' even when stateInfo.mailingAddress === ''
```

Comment:
+1 for considering corner cases.  But `stateInfo.mailingAddress` never includes 'above' as in 'Same as above'.  However, 'above' may show up as a random substring inside an address, which would lead to an error.  I think you meant to only check `!stateInfo.mailingAddress`.